src/NuGet.Core/NuGet.Common/Telemetry/TelemetryActivity.cs

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace NuGet.Common
+{
+    public class TelemetryActivity : IDisposable
+    {
+        private DateTimeOffset _startTime;
+        private Stopwatch _stopwatch;
+        private Stopwatch _intervalWatch = new Stopwatch();
+        private List<Tuple<string, TimeSpan>> _intervalList;
+
+        public TelemetryEvent TelemetryEvent { get; set;}
+
+        public KeyValuePair<string, Guid> ParentId { get; }
+
+        public static INuGetTelemetryService NuGetTelemetryService { get; set; }
+
+        public TelemetryActivity(KeyValuePair<string, Guid> parentId) :
+            this(parentId, null)
+        {
+        }
+
+        public TelemetryActivity(KeyValuePair<string, Guid> parentId, TelemetryEvent telemetryEvent)
```

Comment:
you should also add a operation id here so that everytime a TelemetryActivity started, it generates a new operation id for that.

Reply:
no, not all ids are opeartionId